1Password/docusaurus-extensions
# [DEV] Upgrade Jest packages to latest versions

## Summary

Upgrade Jest and adjacent packages to their latest versions.

## Related Issues and MRs

Closing #59 

## What changed

- Upgrade `jest` and `jest-environment-jsdom` to 29.7.0
- Upgrade `@types/jest` to 29.5.12
- Upgrade `@jest/types` to 29.6.3
- Upgrade `ts-jest` to 28.0.8
- Upgrade `eslint-plugin-jest` to 28.2.0
- Upgrade `@testing-library/jest-dom` to `6.4.2`

## Steps for reviewer

1. Pull changes and install
```
git checkout brian/jest
yarn install
```
2. Verify 'jest' packages have been updated throughout the project
3. Ensure tests are still passing. Can use CI as verification
```
yarn test
```
4. Ensure plugins compile successfully
```
yarn dev docusaurus-plugin-stored-data
```
```
yarn dev jest-docusaurus
```

## Additional resources

N/A
